[FLINK-28088] Adjust table store document to catalog (#159)
apache/flink-table-store: docs/content/_index.md

# Apache Flink Table Store

Flink Table Store is a unified storage to build dynamic tables for both streaming and

> batch processing in Flink, supporting high speed data ingestion and timely data query.

Reviewer: ```suggestion
batch processing in Flink, supporting high-speed data ingestion and timely data query.
```

Table Store offers the following core capabilities:

> - Support storage of large datasets and allows read / write in both batch and streaming manner.

Reviewer: ```suggestion
- Support storage of large datasets and allow read/write in both batch and streaming mode.
```

- Support streaming queries with minimum latency down to milliseconds.
- Support Batch/OLAP queries with minimum latency down to the second level.

> - Support incremental snapshots for stream consumption by default. So users don’t need to solve the problem of combining different stores by themselves.

Reviewer: ```suggestion
- Support incremental snapshots for stream consumption by default. So users do not need to combine different pipelines by themself.
```
